packages/core/core/src/factories/useCartFactory.ts

```diff
@@ -38,20 +38,15 @@ interface UseCartFactory<CART, CART_ITEM, PRODUCT, COUPON> {
 export const useCartFactory = <CART, CART_ITEM, PRODUCT, COUPON>(
   factoryParams: UseCartFactoryParams<CART, CART_ITEM, PRODUCT, COUPON>
 ): UseCartFactory<CART, CART_ITEM, PRODUCT, COUPON> => {
-  const appliedCoupon: Ref<COUPON | null> = ref(null);
-  const loading: Ref<boolean> = ref<boolean>(false);
-  const cart: Ref<CART> = ref(null);
-  let isInitialized = false;
 
   const setCart = (newCart: CART) => {
-    cart.value = newCart;
+    getShared('useCart-cart').value = newCart;
```

Comment:
I really don't like this construction. Basing refs on keys is nether close to the native implementation nro DX-friendly (no autosuggestion, no clear information on from where the import is coming etc).

Also it's misleading since getter is implying a read-only property

 Can we somehow change it to regular imports?

Reply:
yeah.. don't like it too - I have an idea for having just `sharedRef` - so let's improve